src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -98,12 +98,12 @@ default String getUniqueNameForRunnerFile() {
     }
     
     // This method prepares the temp folder by coping all helper files in it.
-    default void prepareTmpFldr(FilePath tmpFldr) throws IOException, InterruptedException {
-        // Copy MATLAB scratch file in temp.
-        copyFileInWorkspace(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_RESOURCE,
-                MatlabBuilderConstants.MATLAB_TEST_RUNNER_FILE_PREFIX
-                        + tmpFldr.getBaseName().replaceAll("-", "_") + ".m",
-                tmpFldr);
+    default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOException, InterruptedException {
+        // Write MATLAB scratch file in temp folder.
+        FilePath scriptFile =
+                new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_TEST_RUNNER_FILE_PREFIX
+                        + tmpFldr.getBaseName().replaceAll("-", "_") + ".m");

```

Comment:
`.getBaseName().replaceAll("-","_")` seems like a common occurrence in the code. You may consider performing this once and storing the result in a variable or putting this in a function or something.

MATLAB has more restriction on script names than just not allowing "-" (you can't start it with a number for instance, see [isvarname](https://www.mathworks.com/help/matlab/ref/isvarname.html)). You happen to know that `tmpFldr` isn't going to get a folder name that starts with a number, but it's best to avoid this kind of shared knowledge throughout the code if possible. You may eventually forget `prepareTmpFldr` has this precondition and new contributors won't know it to begin with.

Reply:
This was good point @mcafaro . Thanks for this perspective. I have extracted this method out in isolation which knows the details of MATLAB specific file names. and others need not be aware of this.